### ISSUE TITLE
No Changelog on APT (Debian) package in Brave repository

### ISSUE THREAD

**Issue by sanawu** [brave/brave-browser#12814]
**Steps to reproduce**
Execute the command: "sudo apt-get changelog brave-browser"

**Actual output**
E: Failed to fetch changelog:/brave-browser.changelog  Changelog unavailable for brave-browser=1.17.73

**Expected output**
THE Changelog text

**Package source repository used**
deb [arch=amd64] https://brave-browser-apt-release.s3.brave.com/ stable main

**Actual output**
"No changelog available"

**Comment by franxois** [comment]
brave-browser-apt-release.s3.brave.com is down ?

**Comment by sanawu** [comment]
> brave-browser-apt-release.s3.brave.com is down ?

I don't believe they've ever published a changelog on the repository source, I've always had to check the website, but the package system has a mechanism to show changelogs and that's what I want to see

**Comment by fizzatbeyond** [comment]
When I execute  "apt-get changelog brave-browser" I get a change log.  Granted it's for 1.16.76.  The real issue appears to be as franxois says, the S3 repository is down.

Err:4 https://brave-browser-apt-release.s3.brave.com stable InRelease          
  Could not resolve 'brave-browser-apt-release.s3.brave.com'

W: Failed to fetch https://brave-browser-apt-release.s3.brave.com/dists/stable/InRelease  Could not resolve 'brave-browser-apt-release.s3.brave.com'

**Comment by sanawu** [comment]
> When I execute "apt-get changelog brave-browser" I get a change log. Granted it's for 1.16.76. The real issue appears to be as franxois says, the S3 repository is down.
> 
> Err:4 https://brave-browser-apt-release.s3.brave.com stable InRelease
> Could not resolve 'brave-browser-apt-release.s3.brave.com'
> 
> W: Failed to fetch https://brave-browser-apt-release.s3.brave.com/dists/stable/InRelease Could not resolve 'brave-browser-apt-release.s3.brave.com'

I have never seen it in mintupdate changelog pane, and that's basically a frontend to apt

**Comment by sanawu** [comment]
Service is up again, still no changelog

No changelog is available via their repository

**Comment by sanawu** [comment]
> When I execute "apt-get changelog brave-browser" I get a change log. Granted it's for 1.16.76. The real issue appears to be as franxois says, the S3 repository is down.
> 
> Err:4 https://brave-browser-apt-release.s3.brave.com stable InRelease
> Could not resolve 'brave-browser-apt-release.s3.brave.com'
> 
> W: Failed to fetch https://brave-browser-apt-release.s3.brave.com/dists/stable/InRelease Could not resolve 'brave-browser-apt-release.s3.brave.com'

I even tried a specific older version

apt-get changelog brave-browser=1.16.76

and I get this

"E: Failed to fetch changelog:/brave-browser.changelog  Changelog unavailable for brave-browser=1.16.76"

**Comment by nognognoggynog** [comment]
I also don't get the changelog via apt

**Comment by bsclifton** [comment]
cc: @fmarier

**Comment by fmarier** [comment]
> I don't believe they've ever published a changelog on the repository source, I've always had to check the website, but the package system has a mechanism to show changelogs and that's what I want to see

That's correct. We've never shipped a proper changelog in the packages (have a look at `/usr/share/doc/brave-browser/changelog.gz` to see that it's rewritten on each release) nor made that available via the repository.

At the moment, it's only available on https://brave.com/latest/.

**Comment by fmarier** [comment]
I do agree that it would be nice to have, but I can't promise it will happen anytime soon since it's going to be a lower priority than fixing other Linux-specific issues.

One of the problems is that Chromium, which is what we use to build our Linux packages doesn't have proper support for generating/including a changelog and so this would require extra patching of the source. It would also be the first platform where we provide the changelog as part of the build process and so it would involve changes to our release process too.

**Comment by fmarier** [comment]
Also, just noting that this is a pretty widespread problem when it comes to third-party repositories:
```
$ apt changelog nodejs
E: Failed to fetch changelog:/nodejs.changelog  Changelog unavailable for nodejs=12.20.1-deb-1nodesource1
$ apt changelog llvm-11
E: Failed to fetch changelog:/llvm-toolchain-11.changelog  Changelog unavailable for llvm-toolchain-11=1:11.1.0~++20210112083037+9bbcb554cdbf-1~exp1~20210112193657.159
$ apt changelog yubikey-manager
E: Failed to fetch changelog:/yubikey-manager.changelog  Changelog unavailable for yubikey-manager=3.1.2~ppa5~focal1
```
